R/parse.R

```diff
@@ -77,7 +78,10 @@ parse_in_half <- function(x, pattern) {
 parse_match <- function(x, pattern) {
   m <- regexec(pattern, x, perl = TRUE)
   pieces <- regmatches(x, m)[[1]][-1]
-  map(pieces, empty_to_null)
+
+  # replace empty element with null
+  pieces[pieces == ""] <- list(NULL)
```

Comment:
This is a bit confusing to me because it implicitly converts the character vector to a list. Maybe this would be clearer?

```R
empty <- pieces == ""
pieces <- as.list(pieces)
pieces[empty] <- list(NULL)
```

Or maybe use `ifelse()`?

```R
ifelse(pieces == "", list(NULL), as.list(pieces))
```

Would you mind seeing if those approaches work and what the impact on the benchmarks is?

Reply:
```r
empty <- pieces == ""
pieces <- as.list(pieces)
pieces[empty] <- list(NULL)
```
Had a small (hardly noticeable) impact on performance to the implicit conversion of character vector to a list.


```r
ifelse(pieces == "", list(NULL), as.list(pieces))
```
Sadly had a larger impact and gave performance similar to the original implementation.

Reply:
Yeah, lets go with the explicit coercion to a list then; I think that's easier to reason about.